src/openforms/registrations/utils.py

```diff
@@ -30,3 +33,34 @@ def execute_unless_result_exists(
     assign(submission.registration_result, spec, result, missing=dict)
     submission.save(update_fields=["registration_result"])
     return callback_result
+
+
+def collect_registrations_failures(since: datetime) -> list[dict[str, str] | None]:
+    logs = TimelineLogProxy.objects.filter(
+        timestamp__gt=since,
+        extra_data__log_event="registration_failure",
+        extra_data__include_in_daily_digest=True,
+    ).order_by("timestamp")
```

Comment:
I think you can do some clever `.annotate` behaviour to get all the necessary information in one go rather than having to post-process in Python:

```py
qs = TimelineLogProxy.objects\
    .filter(...)\
    .values("object_id")\
    .annotate(
        num_failed=Count("object_id"), 
        first_at=Min("timestamp"), 
        last_at=Max("timestamp"),
    )
```
which gives you the raw data, and then you have the form IDs so you can fetch the UUIDs and names:

```py
class FailedSubmission:
    form: Form
    num_failed: int
    first_failure_at: datetime
    last_failure_at: datetime

    @property
    def admin_link(self) -> str:
       ...


forms = Form.objects.in_bulk(id_list=[item["object_id"] for item in qs], field_name="id")
aggregated_logs = [
    FailedSubmission(form=forms[ int(item["object_id"]) ], num_failed=item["num_failed"], ...)
    for item in qs
]
```

Reply:
This was discussed with @sergei-maertens in the office and we decided that we will go for the pythonic way since adding django's `GenericRelation` for this is a pain.